SU2_CFD/src/solver_direct_mean.cpp

```diff
@@ -7320,231 +7308,195 @@ void CEulerSolver::SetActDisk_BCThrust(CGeometry *geometry, CSolver **solver_con
-    
-    /*--- Compute a new value for AoA on the fine mesh only (radians)---*/
-    
-    if (iMesh == MESH_0) AoA = AoA_old + AoA_inc;
-    else { AoA = config->GetAoA()*PI_NUMBER/180.0; }
-    
-    /*--- Only the fine mesh stores the updated values for AoA in config ---*/
-    
-    if (iMesh == MESH_0) {
-      config->SetAoA(AoA*180.0/PI_NUMBER);
-    }
-    
-    /*--- Update the freestream velocity vector at the farfield ---*/
-    
-    for (iDim = 0; iDim < nDim; iDim++)
-      Vel_Infty[iDim] = GetVelocity_Inf(iDim);
-    
-    /*--- Compute the magnitude of the free stream velocity ---*/
-    
-    Vel_Infty_Mag = 0;
-    for (iDim = 0; iDim < nDim; iDim++)
-      Vel_Infty_Mag += Vel_Infty[iDim]*Vel_Infty[iDim];
-    Vel_Infty_Mag = sqrt(Vel_Infty_Mag);
-    
-    /*--- Compute the new freestream velocity with the updated AoA ---*/
-    
-    if (nDim == 2) {
-      Vel_Infty[0] = cos(AoA)*Vel_Infty_Mag;
-      Vel_Infty[1] = sin(AoA)*Vel_Infty_Mag;
-    }
-    if (nDim == 3) {
-      Vel_Infty[0] = cos(AoA)*cos(Beta)*Vel_Infty_Mag;
-      Vel_Infty[1] = sin(Beta)*Vel_Infty_Mag;
-      Vel_Infty[2] = sin(AoA)*cos(Beta)*Vel_Infty_Mag;
-    }
-    
-    /*--- Store the new freestream velocity vector for the next iteration ---*/
-    
-    for (iDim = 0; iDim < nDim; iDim++) {
-      Velocity_Inf[iDim] = Vel_Infty[iDim];
-    }
-    
-    /*--- Only the fine mesh stores the updated values for velocity in config ---*/
-    
-    if (iMesh == MESH_0) {
-      for (iDim = 0; iDim < nDim; iDim++)
-        config->SetVelocity_FreeStreamND(Vel_Infty[iDim], iDim);
+  AoA = config->GetAoA();
+
+  /* --- If AoA needs to be updated, calculate change in AoA --- */
+
+  if (Update_AoA && Output) {
+
+    /*--- Estimate the increment in AoA based on dCL_dAlpha (degrees) ---*/
+    if (!CL_Converged){
+      AoA_inc = (1.0/dCL_dAlpha)*(Target_CL - Total_CL);
     }
-    
+
     /*--- Output some information to the console with the headers ---*/
     
-    if ((rank == MASTER_NODE) && (iMesh == MESH_0) && write_heads && !config->GetDiscrete_Adjoint()) {
-      Old_AoA = config->GetAoA() - AoA_inc*(180.0/PI_NUMBER);
-      
+    if ((rank == MASTER_NODE) && (iMesh == MESH_0) && !config->GetDiscrete_Adjoint()) {
       cout.precision(7);
       cout.setf(ios::fixed, ios::floatfield);
       cout << endl << "----------------------------- Fixed CL Mode -----------------------------" << endl;
       cout << "CL: " << Total_CL;
       cout << " (target: " << config->GetTarget_CL() <<")." << endl;
-      cout.precision(4);
-      cout << "Previous AoA: " << Old_AoA << " deg";
-      cout << ", new AoA: " << config->GetAoA() << " deg." << endl;
+      if (CL_Converged){
+        cout << "CL converged to within "<< config->GetCauchy_Eps() <<", not changing AoA" << endl;
+      }
+      else {
+        cout.precision(4);
+        cout << "Previous AoA: " << AoA << " deg";
+        cout << ", new AoA: " << AoA + AoA_inc << " deg." << endl;
+      }
```

Comment:
Can you add all these things as history fields to the output structure ?

Reply:
Yup! I was thinking of doing that as well. Angle of attack is already at output field, so the only thing left is the C_L convergence. I will add that. 

Did you also mean to add the print statements like `Previous AoA:`?

Reply:
Yeah I mean all of the values. Maybe we can also remove that output then ?

Reply:
I can add Previous AoA to the output_structure. But I do think that having a little header when it actually changes the AoA is useful.